homeassistant/components/monzo/__init__.py

```diff
@@ -0,0 +1,210 @@
+"""
+Support for Monzo accounts.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/monzo/
+"""
+from datetime import timedelta
+import logging
+import time
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_MONITORED_CONDITIONS, CONF_SCAN_INTERVAL, CONF_SENSORS)
+from homeassistant.core import callback
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+
+# We need an import from .config_flow, without it .config_flow is never loaded.
+from .config_flow import configured_instances  # noqa: F401
+from .const import (
+    CONF_ACCESS_TOKEN, CONF_CLIENT_ID, CONF_CLIENT_SECRET, CONF_EXPIRES_AT,
+    CONF_LAST_SAVED_AT, CONF_REFRESH_TOKEN, DOMAIN)
+
+REQUIREMENTS = ['monzo==0.8.0']
+
+_CONFIGURING = {}
+_LOGGER = logging.getLogger(__name__)
+
+
+DATA_MONZO_CLIENT = 'data_client'
+DATA_MONZO_LISTENER = 'data_listener'
+DATA_MONZO_CONFIG = 'monzo_config'
+DATA_BALANCE = 'balance'
+DATA_POTS = 'pots'
+DATA_ACCOUNTID = 'account_id'
+
+DEFAULT_ATTRIBUTION = 'Data provided by Monzo'
+DEFAULT_SCAN_INTERVAL = timedelta(minutes=5)
+
+TOPIC_UPDATE = '{0}_data_update'.format(DOMAIN)
+
+TYPE_BALANCE = 'balance'
+TYPE_DAILY_SPEND = 'dailyspend'
+TYPE_POTS = 'pots'
+
+SENSORS = {
+    TYPE_BALANCE: ['Account Balance', 'mdi:cash', 'GBP'],
+    TYPE_DAILY_SPEND: ['Spent Today', 'mdi:cash', 'GBP'],
+    TYPE_POTS: ['Balance', 'mdi:cash', 'GBP']
+}
+
+SENSOR_SCHEMA = vol.Schema({
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=list(SENSORS)):
+        vol.All(cv.ensure_list, [vol.In(SENSORS)])
+})
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_CLIENT_ID): cv.string,
+        vol.Required(CONF_CLIENT_SECRET): cv.string,
+        vol.Optional(CONF_SENSORS): SENSOR_SCHEMA,
+        vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL):
+            cv.time_period,
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass, config):
+    """Set up Monzo components."""
+    hass.data[DOMAIN] = {}
+    hass.data[DOMAIN][DATA_MONZO_CLIENT] = {}
+    hass.data[DOMAIN][DATA_MONZO_LISTENER] = {}
+
+    if DOMAIN not in config:
+        return True
+
+    conf = config[DOMAIN]
+    client_id = conf.get(CONF_CLIENT_ID)
```

Comment:
Use `dict[key]` for required config keys and keys with default config schema values.